libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -69,6 +75,30 @@ const AP_Param::GroupInfo AP_Baro::var_info[] = {
     // @Description: This selects which barometer will be the primary if multiple barometers are found
     // @Values: 0:FirstBaro,1:2ndBaro,2:3rdBaro
     AP_GROUPINFO("PRIMARY", 6, AP_Baro, _primary_baro, 0),
+    
+    // @Param: ADJ_RATE
+    // @DisplayName: baro adjustment rate
+    // @Description: Rate of barometer adjustment using GPS. If enabled (>=0.1), the barometer ground pressure is slowly adjusted during flight segments with constant altitude target by using GPS information.  Recommended value: 1cm/sec
+    // @Units: cm/second
+    // @Range: 0 10
+    // @Increment: 0.1
+    AP_GROUPINFO("ADJ_RATE", 7, AP_Baro, _adj_step, ADJ_RATE_DEFAULT),
+
+    // @Param: ADJ_TC
+    // @DisplayName: GPS filter time constant
+    // @Description: Time constant for the low pass filter applied to the GPS signal. As we only want to compensate for long-term trends, the should be fairly large (30sec...several mintes)
+    // @Units: seconds
+    // @Range: 10 600
+    // @Increment: 1
+    AP_GROUPINFO("ADJ_TC", 8, AP_Baro, _adj_timeconstant, ADJ_TC_DEFAULT),
```

Comment:
I think 1 minute is still quite aggressive, when going conservatively two, three or even 5 minutes would be better. The problem with that choice: the filter needs to be fed at least the same time before the correction output makes sense. 

My use case is a >1h copter flight in AUTO (mapping), so I used to set 180 seconds here, together with 300 seconds of delay (maybe overly conservative, but it worked well). But I guess most people  with copters would run low on batteries before the adjustment starts after 5 minutes. Personally, I would keep it configurable therefore ...